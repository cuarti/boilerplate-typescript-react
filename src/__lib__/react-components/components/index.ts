

export * from './Button';
export * from './GroupButton';
