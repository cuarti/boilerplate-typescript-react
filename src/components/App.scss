

.app {
	color: #333;
	text-align: center;
}

.foo-bar {
	color: orangered;
}
