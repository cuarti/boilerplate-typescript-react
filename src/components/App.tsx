
import * as React from 'react';

import {ImageButton} from './ImageButton';

// import * as style from './App.scss';
const style = require('./App.scss');


export class App extends React.Component<{}, {}> {

	public render(): JSX.Element {
		return (
			<div className={style.app}>
				<h1>boilerplate-typescript-react</h1>
				<p>Full boilerplate for react in typescript with webpack2 and hot module replacement</p>
				<ImageButton text="Click-me!" img="/images/tux.png" />
			</div>
		);
	}

}
