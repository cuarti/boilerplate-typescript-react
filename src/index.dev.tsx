
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {AppContainer} from 'react-hot-loader';

import {App} from './components/App';


declare let module: {hot: any};


const root = document.getElementById('root');

ReactDOM.render(
	<AppContainer>
		<App/>
	</AppContainer>,
	root
);

if(module.hot) {
	module.hot.accept('./components/App', () => {

		const NewApp = require('./components/App').App;

		ReactDOM.render(
			<AppContainer>
				<NewApp/>
			</AppContainer>,
			root
		);
	});
}
