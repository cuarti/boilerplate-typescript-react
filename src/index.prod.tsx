
import * as React from 'react';
import * as ReactDOM from 'react-dom';

import {App} from './components/App';


const root = document.getElementById('root');

ReactDOM.render(
	<App/>,
	root
);
