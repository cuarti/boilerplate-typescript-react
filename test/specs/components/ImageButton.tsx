
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {ImageButton} from '../../../src/components';


test('', () => {

	let component = renderer.create(
		<ImageButton text="click me!"
					 img="/images/tux.png" />
	);

	expect(component.toJSON().children)
		.toEqual([
			{
				type: 'img',
				props: { className: 'image', src: '/images/tux.png' },
				/* tslint:disable-next-line:no-null-keyword */
				children: null
			},
			{
				type: 'span',
				props: { className: 'text' },
				children: [ 'click me!' ]
			}]
		);

	expect(component.toJSON()).toMatchSnapshot();

});
