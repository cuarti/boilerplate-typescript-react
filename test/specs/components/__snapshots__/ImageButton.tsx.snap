// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className={undefined}
  onClick={undefined}
>
  <img
    className="image"
    src="/images/tux.png"
  />
  <span
    className="text"
  >
    click me!
  </span>
</button>
`;
